python/tests/tsutil.py

```diff
@@ -1713,6 +1715,196 @@ def iterate(self):
             left = right
 
 
+FORWARD = 1
+REVERSE = -1
+
+
+@dataclasses.dataclass
+class Interval:
```

Comment:
Why not use the existing `Interval` class?

Reply:
Just simpler to keep track of what it does for this kind of prototype/test code, and not have to worry about external APIs and so on